modules/ags/src/main/scala/lucuma/ags/Ags.scala

```diff
@@ -97,35 +98,57 @@ object Ags {
 
   }
 
+  private def offsetAt(
+    at:      Instant => Option[Coordinates],
+    instant: Instant,
+    gsc:     GuideStarCandidate
+  ): Option[Offset] =
+    (at(instant), gsc.tracking.at(instant)).mapN(_.diff(_).offset)
+
+  private def scienceOffsetsAt(
+    scienceAt: List[Instant => Option[Coordinates]],
+    instant:   Instant,
+    gsc:       GuideStarCandidate
+  ): List[Offset] =
+    scienceAt
+      .map(s => offsetAt(s, instant, gsc))
+      .collect { case Some(x) =>
+        x
+      }
```

Comment:
`.flatten`

Reply:
I recall we stopped using `flatten` for this case instead of collect @tpolecat Had a good reason